src/cpu/core_dynrec/operators.h

```diff
@@ -1967,6 +1967,152 @@ static uint32_t DRC_CALL_CONV dynrec_stosd_dword(uint32_t count,int32_t add_inde
 	return count_left;
 }
 
+static uint16_t DRC_CALL_CONV dynrec_scasb_word(uint16_t count, int16_t add_index,
+                                                PhysPt di_base) DRC_FC;
+static uint16_t DRC_CALL_CONV dynrec_scasb_word(uint16_t count,
+                                                int16_t add_index, PhysPt di_base)
+{
+	for (; count > 0; count--) {
+		const auto addr = di_base + reg_di;
+		const auto val  = mem_readb(addr);
+
+		lf_var1b    = reg_al;
+		lf_var2b    = val;
+		lf_resb     = (uint8_t)(lf_var1b - lf_var2b);
+		lflags.type = t_CMPb;
+
+		reg_di += add_index;
+		if (lf_resb == 0) {
+			count--;
+			break;
+		}
+	}
+
+	return count;
```

Comment:
Just an idea; could `return count` in the `if` statement, and then `return 0` at the end because `count` should always be zero at that point.

Reply:
I myself am a bigger fan of single return statements in a function unless it causes major contortions.

Reply:
I'm generally the same, yeah. @kcgen has convinced me to prefer returning early vs nesting the whole function body, but in general I dislike multiple return statements too.

Reply:
yup; all good!
When we've got a waterfall of chunks of logic, the early returns can help flatten or act like a hard partition, which can be easy to juggle stuff in your head.
But yeah - in this case, it just collapsed down to a single return statement.

Was thinking the interesting aspect here might be to `return 0` at the end as a way to self-document/explain that zero is always expected if we get to the end of the function. (Or another way: the only time we return a non-zero `count` is when `lf_resb == 0`.